src/Tribe/iCal.php

```diff
@@ -359,9 +360,31 @@ public static function generate_ical_feed( $post = null ) {
 		$content = apply_filters( 'tribe_ical_properties', $content );
 		$content .= $events;
 		$content .= 'END:VCALENDAR';
-		echo $content;
 
-		exit;
+		if ( $echo ) {
+			echo $content;
```

Comment:
copied and pasted really, joining this suggestion and Barry's one about the use of `tribe_exit`